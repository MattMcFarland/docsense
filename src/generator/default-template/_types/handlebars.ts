export interface BlockHelper {
  fn: (ctx: any) => any;
}
