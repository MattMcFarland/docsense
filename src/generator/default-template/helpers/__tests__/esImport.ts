import * as Handlebars from 'handlebars';

describe('hbs helpers > esImport', () => {
  test('when all is false > matches snapshot', () => {
    Handlebars.registerHelper('esImport', require('../esImport'));
    const template = Handlebars.compile(`
    {{ esImport
      all=false
      name="foo"
      link="/bar/baz/source.html"
      line=99
      from="/bar/baz"
    }}
    `);
    const result = template({});
    expect(result).toMatchSnapshot();
  });
  test('when all is true > matches snapshot', () => {
    Handlebars.registerHelper('esImport', require('../esImport'));
    const template = Handlebars.compile(`
    {{ esImport
      all=true
      name="foo"
      link="/bar/baz/source.html"
      line=99
      from="/bar/baz"
    }}
    `);
    const result = template({});
    expect(result).toMatchSnapshot();
  });
});
