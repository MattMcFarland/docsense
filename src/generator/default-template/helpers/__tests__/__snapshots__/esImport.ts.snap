// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hbs helpers > esImport when all is false > matches snapshot 1`] = `
"
    <pre class=\\"ba b--black-20 lh-solid\\"><code class=\\"javascript hljs\\"><span class=\\"hljs-keyword\\">import</span> <span>foo</span> <span class=\\"hljs-keyword\\">from</span> <a title=\\"view source\\" class=\\"dim dark-red link\\" href=\\"/bar/baz/source.html#99\\">'docsense/bar/baz'</a></code></pre>
    "
`;

exports[`hbs helpers > esImport when all is true > matches snapshot 1`] = `
"
    <pre class=\\"ba b--black-20 lh-solid\\"><code class=\\"javascript hljs\\"><span class=\\"hljs-keyword\\">import</span> * <span class=\\"hljs-keyword\\">as</span> <span>foo</span> <span class=\\"hljs-keyword\\">from</span> <a title=\\"view source\\" class=\\"dim dark-red link\\" href=\\"/bar/baz/source.html#99\\">'docsense/bar/baz'</a></code></pre>
    "
`;
